src/js/structs/Batch.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * An immutable batch is an ever growing batch with a reduce capability.
+ *
+ * This can be used to keep action transactions that can be applied through
+ * a set of reducers to a data set in order to transform it.
+ *
+ * Why 'Batch'? Because it's not really a `Stack`, since you cannot `pop`
+ * items from it. Also not a `Log`, because it can be confused with the
+ * logging utilities.
+ *
+ * @example <caption>Using Batch</caption>
+ * class Component extends React.Component() {
+ *   constructor() {
+ *     super();
+ *     this.state = {
+ *        actionBatch: new Batch()
+ *     }
+ *   }
+ *
+ *   handleAction() {
+ *     this.setState({
+ *        actionBatch: this.state.actionBatch.add({
+ *           action: 'set',
+ *           key: 'foo',
+ *           value: 'bar'
+ *        })
+ *     })
+ *   }
+ *
+ *   render() {
+ *      let data = this.state.actionBatch.reduce(
+ *            applyActionsToDataCallback,
+ *            this.props.data
+ *         );
+ *
+ *      return <Visualize data={data} />
+ *   }
+ * }
+ *
+ *
+ */
+
+class Batch {
+
+  constructor() {
+    const context = {instance:this, batch:[]};
```

Comment:
I'll add a comment

``
// This creates a context for the functions which have access to the batch variable and also 
// access to the instance. So `this` can be returned and functions could be chained.
``

This is also necessary, if one extends the batch and overwrites the add method to do some filtering.